userspace/libsinsp/parsers.cpp

```diff
@@ -422,20 +422,24 @@ void sinsp_parser::process_event(sinsp_evt *evt) {
 	case PPME_SYSCALL_MKDIRAT_X:
 	case PPME_SYSCALL_UNLINKAT_2_X:
 	case PPME_SYSCALL_MKNODAT_X: {
-		auto res = evt->get_param(0)->as<int64_t>();
-		if(res >= 0) {
-			// Only if successful
-			auto dirfd = evt->get_param(1)->as<int64_t>();
-			evt->set_fd_info(evt->get_tinfo()->get_fd(dirfd));
+		if (evt->get_tinfo() != nullptr) {
```

Comment:
[nitpick] This is very go-looking and fine, but I generally prefer early returns, in this particular case it would result in less lines changed and generally better readability by reducing the amount of nesting. But of course, this is just a personal preference.

```suggestion
		if (evt->get_tinfo() == nullptr) {
			break;
		}
```

Reply:
Agreed :)